x-pack/platform/plugins/shared/security/server/authentication/providers/saml.ts

```diff
@@ -885,4 +881,14 @@ export class SAMLAuthenticationProvider extends BaseAuthenticationProvider {
       { state: state ? state : null }
     );
   }
+
+  /**
+   * Determines whether the provided token is a UIAM token. Generally, we shouldn't rely on the
+   * shape of the UIAM tokens to trigger UIAM-specific logic, but this is necessary for the
+   * transition period while we support both SAML and UIAM tokens at the same time.
+   * @param token ES native or UIAM access or refresh token.
+   */
+  private isUiamToken(token?: string): this is { options: { uiam: UiamServicePublic } } {
+    return this.useUiam && !!token?.startsWith('essu_');
```

Comment:
Question: is it worth adding logging for a scenario (like the QA issue recently 😅 ) Where we have a UIAM access token, but are not running in UIAM mode?

Or should we consider that unlikely since we will only go from "standard mode" to "uiam mode", and not the reverse?